src/appleseed/renderer/modeling/project/project.xsd

```diff
@@ -399,6 +399,9 @@
                         <xsd:complexType>
                             <xsd:complexContent>
                                 <xsd:extension base="parameterSet">
+                                    <xsd:choice minOccurs="0" maxOccurs="1">
+                                        <xsd:element name="osl_code"/>

```

Comment:
What about a more widely applicable `<code>` element with a language attribute?

    <code language="osl">
        ...
    </code>

We need to be extra careful with schema changes since migrations are particularly painful...

Reply:
osl_code is not an entity that we reference in other entities. It's inlined inside OSL shaders.
`
 <shader type="shader" name="source_code" layer="source_code1">
    <parameter name="in_freq" value="float 151" />
        <osl_code>                                                            
            shader sinxgradient(float in_freq = 0.0, output float out_value = 0.0)
            {
                out_value = sin(P[0] * in_freq);
            }
        </osl_code>
    </shader>
`

Reply:
I know, what I'm saying is that I would prefer

    <code language="osl">
        ...
    </code>

to

    <osl_code>
        ...
    </osl_code>

Reply:
Or, if we assume that `<shader>` can only ever use OSL shaders, then simply

    <code>
        ...
    </code>

Reply:
Here we will only support OSL code. If we ever add another shading language to appleseed, it will be another entity.

I prefer <osl_code>. It makes clear what the tag contains and we maybe want to use a more generic <code> tag later for something else.